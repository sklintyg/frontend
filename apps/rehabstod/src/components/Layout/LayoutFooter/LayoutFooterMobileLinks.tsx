import { useGetLinksQuery } from '../../../store/api'
import { useAppDispatch } from '../../../store/hooks'
import { updateShowCookieDialog } from '../../../store/slices/cookieDialog.slice'

export function LayoutFooterMobileLinks() {
  const { data: links } = useGetLinksQuery()
  const dispatch = useAppDispatch()

  return (
    <div className="ids-footer-1177-admin__mobile-links">
      {links?.ineraBehandlingPersonuppgifter && (
        <a className="ids-link ids-link--icon ids-link--small ids-link--underlined" href={links.ineraBehandlingPersonuppgifter.url}>
          {links.ineraBehandlingPersonuppgifter.text}
          <span className="ids-icon-external-link-small ids-icon--text-end" />
        </a>
      )}
      <button
        type="button"
        onClick={() => dispatch(updateShowCookieDialog(true))}
        className="ids-link ids-link--icon ids-link--small ids-link--underlined"
      >
        Hantering av kakor
      </button>
    </div>
  )
}
