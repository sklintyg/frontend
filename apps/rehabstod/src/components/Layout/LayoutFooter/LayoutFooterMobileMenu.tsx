import { useGetLinksQuery } from '../../../store/api'

export function LayoutFooterMobileMenu() {
  const { data: links } = useGetLinksQuery()

  return (
    <div className="ids-footer-1177-admin__mobile-menu">
      <nav className="ids-mobile-menu ids-mobile-menu--variation-2" aria-label="mobile-menu">
        <ul>
          {[links?.ineraManualRehabstod, links?.ineraNationellKundservice, links?.rehabstodTillganglighetsredogorelse].map((link) =>
            link ? (
              <li key={link.url} className="ids-mobile-menu-item">
                <div className="ids-mobile-menu-item__inner">
                  <a href={link.url} target={link.target}>
                    {link.text}
                    {link.target === '_blank' && <span className="ids-icon-external-link ids-icon--text-end ml-1" />}
                  </a>
                </div>
              </li>
            ) : null
          )}
        </ul>
      </nav>{' '}
    </div>
  )
}
