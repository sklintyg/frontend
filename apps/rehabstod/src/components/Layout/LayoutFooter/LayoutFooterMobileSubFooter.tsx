import { useGetLinksQuery } from '../../../store/api'
import { DynamicLink } from '../../DynamicLink/DynamicLink'

export function LayoutFooterMobileSubFooter() {
  const { data: links } = useGetLinksQuery()

  return (
    <div className="ids-footer-1177-admin__sub-footer-mobile">
      <div className="ids-footer-1177-admin__sub-footer-mobile-icon">
        <div className="ids-footer-1177-admin__sub-footer-mobile-service-name">Rehabstöd</div>
      </div>
      <div className="ids-footer-1177-admin__sub-footer-mobile-text">
        <p>
          <span>Rehabstöd drivs av</span> <DynamicLink small underlined link={links?.ineraMainPage} />
        </p>
      </div>
    </div>
  )
}
