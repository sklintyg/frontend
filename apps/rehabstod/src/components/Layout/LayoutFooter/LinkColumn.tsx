import type { Link } from '../../../schemas'
import { DynamicLink } from '../../DynamicLink/DynamicLink'

export function LinkColumn({ links }: { links: (Link | undefined)[] }) {
  return (
    <div className="ids-footer-1177-admin__link-col ids-footer-1177-admin__link-col--2 ids-footer-1177-admin__col-size-3">
      <ul>
        {links.map((link) =>
          link ? (
            <li key={link.key}>
              <DynamicLink link={link} arrow colorPreset={1} />
            </li>
          ) : null
        )}
      </ul>
    </div>
  )
}
