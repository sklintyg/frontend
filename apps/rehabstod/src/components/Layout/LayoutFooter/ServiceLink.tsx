import { IDSLink } from '@inera/ids-react'
import { useGetLinksQuery } from '../../../store/api'

export function ServiceLink({ mobile }: { mobile?: boolean }) {
  const { data: links } = useGetLinksQuery()

  return (
    <p>
      Rehabst√∂d drivs av{' '}
      {links?.ineraMainPage && (
        <IDSLink colorPreset={mobile ? undefined : 4} small underlined>
          <a href={links.ineraMainPage.url} target="_blank" rel="noreferrer">
            {links.ineraMainPage.text}
            <span className="ids-icon-external-link-small ids-icon--text-end" />
          </a>
        </IDSLink>
      )}
    </p>
  )
}
