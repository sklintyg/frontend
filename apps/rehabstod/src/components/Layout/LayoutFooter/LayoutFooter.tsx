import '@inera/ids-design/components/footer-1177-admin/footer-1177-admin.css'
import { useGetLinksQuery } from '../../../store/api'
import { useAppDispatch } from '../../../store/hooks'
import { updateShowCookieDialog } from '../../../store/slices/cookieDialog.slice'
import { CookieDialog } from '../../dialog/CookieDialog'
import { DynamicLink } from '../../DynamicLink/DynamicLink'
import { LayoutFooterMobileLinks } from './LayoutFooterMobileLinks'
import { LayoutFooterMobileMenu } from './LayoutFooterMobileMenu'
import { LayoutFooterMobileSubFooter } from './LayoutFooterMobileSubFooter'

export function LayoutFooter() {
  const { data: links } = useGetLinksQuery()
  const dispatch = useAppDispatch()

  return (
    <footer className="ids-footer-1177-admin">
      <CookieDialog />
      <div className="ids-footer-1177-admin__inner-wrapper">
        <div className="ids-footer-1177-admin__inner">
          <div className="ids-footer-1177-admin__headline-row">
            <h1 className="ids-footer-1177-admin__headline">Rehabstöd</h1>
          </div>
          <div className="ids-footer-1177-admin__content">
            <div className="ids-footer-1177-admin__text">
              <p>Rehabstöd används för att samordna och följa upp sjukskrivna patienters rehabilitering.</p>
            </div>

            <div className="ids-footer-1177-admin__link-col">
              <ul>
                <li>
                  <DynamicLink arrow colorpreset={1} link={links?.ineraManualRehabstod} />
                </li>
                <li>
                  <DynamicLink arrow colorpreset={1} link={links?.ineraNationellKundservice} />
                </li>
              </ul>
            </div>

            <div className="ids-footer-1177-admin__link-col">
              <ul>
                <li>
                  <DynamicLink arrow colorpreset={1} link={links?.rehabstodTillganglighetsredogorelse} />
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>

      <div className="ids-footer-1177-admin__sub-footer">
        <div className="ids-footer-1177-admin__sub-footer-container">
          <div className="ids-footer-1177-admin__sub-footer-row">
            <div className="ids-footer-1177-admin__sub-footer-left">
              <p>
                <span>Rehabstöd drivs av</span> <DynamicLink small underlined light link={links?.ineraMainPage} />
              </p>
            </div>
            <div className="ids-footer-1177-admin__sub-footer-right">
              {links?.ineraBehandlingPersonuppgifter && (
                <a className="ids-link ids-link--light" href={links.ineraBehandlingPersonuppgifter.url}>
                  {links.ineraBehandlingPersonuppgifter.text}
                  <span className="ids-icon-external-link-small ids-icon--text-end" />
                </a>
              )}
              <button type="button" onClick={() => dispatch(updateShowCookieDialog(true))} className="ids-link ids-link--light">
                Hantering av kakor
              </button>
            </div>

            <LayoutFooterMobileMenu />
            <LayoutFooterMobileLinks />
            <LayoutFooterMobileSubFooter />
          </div>
        </div>
      </div>
    </footer>
  )
}
