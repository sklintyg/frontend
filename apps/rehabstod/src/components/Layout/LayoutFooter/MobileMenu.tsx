import { TertiaryButton } from '@frontend/components'
import { useGetLinksQuery } from '../../../store/api'
import { useAppDispatch } from '../../../store/hooks'
import { updateShowCookieDialog } from '../../../store/slices/cookieDialog.slice'
import { DynamicLink } from '../../DynamicLink/DynamicLink'
import { ServiceLink } from './ServiceLink'

export function MobileMenu() {
  const { data: links } = useGetLinksQuery()
  const dispatch = useAppDispatch()

  return (
    <>
      <div className="ids-footer-1177-admin__mobile-menu">
        <nav className="ids-mobile-menu ids-mobile-menu--variation-2" aria-label="mobile-menu">
          <ul>
            {[links?.ineraManualRehabstod, links?.ineraNationellKundservice, links?.rehabstodTillganglighetsredogorelse].map((link) =>
              link ? (
                <li key={link.url} className="ids-mobile-menu-item">
                  <div className="ids-mobile-menu-item__inner">
                    <a href={link.url} target={link.target}>
                      {link.text}
                      {link.target === '_blank' && <span className="ids-icon-external-link ids-icon--text-end ml-1" />}
                    </a>
                  </div>
                </li>
              ) : null
            )}
          </ul>
        </nav>
      </div>
      <div className="ids-footer-1177-admin__mobile-links">
        <DynamicLink link={links?.ineraBehandlingPersonuppgifter} />
        <TertiaryButton onClick={() => dispatch(updateShowCookieDialog(true))}>Hantering av kakor</TertiaryButton>
      </div>
      <div className="ids-footer-1177-admin__sub-footer-mobile">
        <div className="ids-footer-1177-admin__sub-footer-mobile-icon">
          <div className="ids-footer-1177-admin__sub-footer-mobile-service-name">Rehabstod</div>
        </div>
        <div className="ids-footer-1177-admin__sub-footer-mobile-text">
          <ServiceLink mobile />
        </div>
      </div>
    </>
  )
}
