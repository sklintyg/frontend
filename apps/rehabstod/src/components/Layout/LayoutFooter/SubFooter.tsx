import { TertiaryButton } from '@frontend/components'
import { useGetLinksQuery } from '../../../store/api'
import { useAppDispatch } from '../../../store/hooks'
import { updateShowCookieDialog } from '../../../store/slices/cookieDialog.slice'
import { DynamicLink } from '../../DynamicLink/DynamicLink'
import { MobileMenu } from './MobileMenu'
import { ServiceLink } from './ServiceLink'

export function SubFooter() {
  const { data: links } = useGetLinksQuery()
  const dispatch = useAppDispatch()

  return (
    <div className="ids-footer-1177-admin__sub-footer">
      <div className="ids-footer-1177-admin__sub-footer-container">
        <div className="ids-footer-1177-admin__sub-footer-row">
          <div className="ids-footer-1177-admin__sub-footer-left">
            <ServiceLink />
          </div>
          <div className="ids-footer-1177-admin__sub-footer-right">
            <DynamicLink link={links?.ineraBehandlingPersonuppgifter} small colorPreset={4} />
            <TertiaryButton small colorPreset={4} onClick={() => dispatch(updateShowCookieDialog(true))} className="ml-5">
              Hantering av kakor
            </TertiaryButton>
          </div>
          <MobileMenu />
        </div>
      </div>
    </div>
  )
}
