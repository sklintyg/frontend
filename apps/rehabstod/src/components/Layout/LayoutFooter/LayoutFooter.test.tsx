import { render, waitFor } from '@testing-library/react'
import { Provider } from 'react-redux'
import { createMemoryRouter, createRoutesFromElements, Route, RouterProvider } from 'react-router-dom'
import { api } from '../../../store/api'
import { store } from '../../../store/store'
import { LayoutFooter } from './LayoutFooter'

it('Should render footer as expected', async () => {
  store.dispatch(api.endpoints.getLinks.initiate())

  await waitFor(() => expect(api.endpoints.getLinks.select()(store.getState()).isSuccess).toBe(true))

  const { container } = render(
    <Provider store={store}>
      <RouterProvider router={createMemoryRouter(createRoutesFromElements(<Route path="/" element={<LayoutFooter />} />))} />
    </Provider>
  )

  expect(container).toMatchSnapshot()
})
