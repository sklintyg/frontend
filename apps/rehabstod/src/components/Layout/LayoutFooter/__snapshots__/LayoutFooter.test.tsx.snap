// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Should render footer as expected 1`] = `
<div>
  <footer
    class="ids-footer-1177-admin print:hidden"
  >
    <div
      class="ids-footer-1177-admin__inner-wrapper"
    >
      <div
        class="ids-footer-1177-admin__inner"
      >
        <div
          class="ids-footer-1177-admin__headline-row"
        >
          <h1
            class="ids-footer-1177-admin__headline"
          >
            Rehabstod
          </h1>
        </div>
        <div
          class="ids-footer-1177-admin__content"
        >
          <div
            class="ids-footer-1177-admin__text"
          >
            <p
              class="ids-body"
            >
              Rehabstöd används för att samordna och följa upp sjukskrivna patienters rehabilitering.
            </p>
          </div>
          <div
            class="ids-footer-1177-admin__link-col ids-footer-1177-admin__link-col--2 ids-footer-1177-admin__col-size-3"
          >
            <ul>
              <li>
                <a
                  class="ids-link ids-link--color-1"
                  href="https://insecure-ophthalmologist.com"
                  rel="noreferrer"
                  target="_blank"
                >
                  <span
                    class="ids-icon-arrow-right-small ids-icon--text-start"
                  />
                  ineraManualRehabstod
                  <span
                    class="ids-icon-external-link-small ids-icon--text-end"
                  />
                </a>
              </li>
              <li>
                <a
                  class="ids-link ids-link--color-1"
                  href="http://wrong-elephant.com"
                  rel="noreferrer"
                  target="_blank"
                >
                  <span
                    class="ids-icon-arrow-right-small ids-icon--text-start"
                  />
                  ineraNationellKundservice
                  <span
                    class="ids-icon-external-link-small ids-icon--text-end"
                  />
                </a>
              </li>
            </ul>
          </div>
          <div
            class="ids-footer-1177-admin__link-col ids-footer-1177-admin__link-col--2 ids-footer-1177-admin__col-size-3"
          >
            <ul>
              <li>
                <a
                  class="ids-link ids-link--color-1"
                  href="http://new-adrenalin.info"
                  rel="noreferrer"
                  target="_blank"
                >
                  <span
                    class="ids-icon-arrow-right-small ids-icon--text-start"
                  />
                  Tillgänglighetsredogörelse
                  <span
                    class="ids-icon-external-link-small ids-icon--text-end"
                  />
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ids-footer-1177-admin__sub-footer"
    >
      <div
        class="ids-footer-1177-admin__sub-footer-container"
      >
        <div
          class="ids-footer-1177-admin__sub-footer-row"
        >
          <div
            class="ids-footer-1177-admin__sub-footer-left"
          >
            <p>
              Rehabstöd drivs av
               
              <a
                class="ids-link ids-link--color-4 ids-link--icon ids-link--light ids-link--small ids-link--underlined"
                href="https://delicious-quadrant.org"
                rel="noreferrer"
                target="_blank"
              >
                ineraMainPage
                <span
                  class="ids-icon-external-link-small ids-icon--text-end"
                />
              </a>
            </p>
          </div>
          <div
            class="ids-footer-1177-admin__sub-footer-right"
          >
            <a
              class="ids-link ids-link--color-4 ids-link--light ids-link--small"
              href="http://grubby-dragster.info"
              rel="noreferrer"
              target="_blank"
            >
              ineraBehandlingPersonuppgifter
              <span
                class="ids-icon-external-link-small ids-icon--text-end"
              />
            </a>
            <button
              class="ids-link inline-flex font-normal items-center ids-link--underlined ids-link--color-4 ids-link--light ids-link--small ml-5"
              type="button"
            >
              <span
                class="ids-link__text"
              >
                Hantering av kakor
              </span>
            </button>
          </div>
          <div
            class="ids-footer-1177-admin__mobile-menu"
          >
            <nav
              aria-label="mobile-menu"
              class="ids-mobile-menu ids-mobile-menu--variation-2"
            >
              <ul>
                <li
                  class="ids-mobile-menu-item"
                >
                  <div
                    class="ids-mobile-menu-item__inner"
                  >
                    <a
                      href="https://insecure-ophthalmologist.com"
                      target="_blank"
                    >
                      ineraManualRehabstod
                      <span
                        class="ids-icon-external-link ids-icon--text-end ml-1"
                      />
                    </a>
                  </div>
                </li>
                <li
                  class="ids-mobile-menu-item"
                >
                  <div
                    class="ids-mobile-menu-item__inner"
                  >
                    <a
                      href="http://wrong-elephant.com"
                      target="_blank"
                    >
                      ineraNationellKundservice
                      <span
                        class="ids-icon-external-link ids-icon--text-end ml-1"
                      />
                    </a>
                  </div>
                </li>
                <li
                  class="ids-mobile-menu-item"
                >
                  <div
                    class="ids-mobile-menu-item__inner"
                  >
                    <a
                      href="http://new-adrenalin.info"
                      target="_blank"
                    >
                      Tillgänglighetsredogörelse
                      <span
                        class="ids-icon-external-link ids-icon--text-end ml-1"
                      />
                    </a>
                  </div>
                </li>
              </ul>
            </nav>
          </div>
          <div
            class="ids-footer-1177-admin__mobile-links"
          >
            <a
              class="ids-link"
              href="http://grubby-dragster.info"
              rel="noreferrer"
              target="_blank"
            >
              ineraBehandlingPersonuppgifter
              <span
                class="ids-icon-external-link-small ids-icon--text-end"
              />
            </a>
            <button
              class="ids-link inline-flex font-normal items-center ids-link--underlined text-base"
              type="button"
            >
              <span
                class="ids-link__text"
              >
                Hantering av kakor
              </span>
            </button>
          </div>
          <div
            class="ids-footer-1177-admin__sub-footer-mobile"
          >
            <div
              class="ids-footer-1177-admin__sub-footer-mobile-icon"
            >
              <div
                class="ids-footer-1177-admin__sub-footer-mobile-service-name"
              >
                Rehabstod
              </div>
            </div>
            <div
              class="ids-footer-1177-admin__sub-footer-mobile-text"
            >
              <p>
                Rehabstöd drivs av
                 
                <a
                  class="ids-link ids-link--icon ids-link--small ids-link--underlined"
                  href="https://delicious-quadrant.org"
                  rel="noreferrer"
                  target="_blank"
                >
                  ineraMainPage
                  <span
                    class="ids-icon-external-link-small ids-icon--text-end"
                  />
                </a>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
